imageloading-core/src/main/java/dev/chrisbanes/accompanist/imageloading/ImageLoad.kt

```diff
@@ -68,78 +64,43 @@ fun <R : Any, TR : Any> ImageLoad(
 ) {
-    var state by remember(requestKey) {
-        // We start from the loading state, to avoid thrashing from empty -> loading straightaway
-        mutableStateOf<ImageLoadState>(ImageLoadState.Loading)
-    }
-
-    val callback by rememberUpdatedState(onRequestCompleted)
-
-    val requestActor = remember(requestKey) {
-        ImageLoadRequestActor(executeRequest)
-    }
+    val updatedOnRequestCompleted by rememberUpdatedState(onRequestCompleted)
+    val updatedTransformRequestForSize by rememberUpdatedState(transformRequestForSize)
+    val updatedExecuteRequest by rememberUpdatedState(executeRequest)
 
-    LaunchedEffect(requestActor) {
-        // Launch the Actor
-        requestActor.run { _, newState ->
-            // Update the result state
-            state = newState
+    var requestSize by remember(requestKey) { mutableStateOf<IntSize?>(null) }
 
-            if (newState is ImageLoadState.Success || newState is ImageLoadState.Error) {
-                callback(newState)
-            }
-        }
+    val loadState by produceState<ImageLoadState>(
+        initialValue = ImageLoadState.Loading,
+        key1 = requestKey,
+        key2 = requestSize,
+        key3 = updatedTransformRequestForSize,
+    ) {
+        value = requestSize?.let { updatedTransformRequestForSize(request, it) }
+            ?.let { transformedRequest ->
+                try {
+                    updatedExecuteRequest(transformedRequest)
+                } catch (throwable: Throwable) {
+                    ImageLoadState.Error(painter = null, throwable = throwable)
+                }.also(updatedOnRequestCompleted)
+            } ?: ImageLoadState.Loading
     }
 
     BoxWithConstraints(
         modifier = modifier,
         propagateMinConstraints = true,
     ) {
-        // We remember the last size in a MutableRef (below) rather than a MutableState.
-        // This is because we don't need value changes to trigger a re-composition, we are only
-        // using it to store the last value.
-        val lastRequestedSize = remember(requestActor) { MutableRef<IntSize?>(null) }
-
-        val requestSize = IntSize(
+        val size = IntSize(
             width = if (constraints.hasBoundedWidth) constraints.maxWidth else -1,
             height = if (constraints.hasBoundedHeight) constraints.maxHeight else -1
         )
 
-        val lastSize = lastRequestedSize.value
-        if (lastSize == null ||
-            (lastSize != requestSize && shouldRefetchOnSizeChange(state, requestSize))
-        ) {
-            val transformedRequest = transformRequestForSize(request, requestSize)
-            if (transformedRequest != null) {
-                requestActor.send(transformedRequest)
-                lastRequestedSize.value = requestSize
-            } else {
-                // If the transform request is null, set our state to empty
-                state = ImageLoadState.Empty
-            }
+        val lastSize = requestSize
```

Comment:
Why do you need `lastSize` as a backing property?

Reply:
No reason, other than being able to keep the next line on 1 line. Removed.